samples/CSharp/SourceGenerators/SourceGeneratorSamples/ResxSourceGenerator.cs

```diff
@@ -0,0 +1,663 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SourceGeneratorSamples
+{
+    [Generator]
+    public class ResxSourceGenerator : ISourceGenerator
+    {
+        public void Initialize(InitializationContext context)
+        {
+            //Debugger.Launch();
+        }
+
+        public void Execute(SourceGeneratorContext context)
+        {
+            IEnumerable<AdditionalText> resourceFiles = context.AdditionalFiles.Where(file => file.Path.EndsWith(".resx", StringComparison.OrdinalIgnoreCase));
+            foreach (AdditionalText resourceFile in resourceFiles)
+            {
+                try
+                {
+                    ProcessResourceFile(context, resourceFile);
+                }
+                catch (Exception ex)
+                {
+                    string[] exceptionLines = ex.ToString().Split(new[] { Environment.NewLine }, StringSplitOptions.None);
+                    string text = string.Join("", exceptionLines.Select(line => "#error " + line + Environment.NewLine));
+                    SourceText errorText = SourceText.From(text, Encoding.UTF8, SourceHashAlgorithm.Sha256);
+                    context.AddSource($"{Path.GetFileName(resourceFile.Path)}.Error", errorText);
+                }
+            }
+        }
+
+        private void ProcessResourceFile(SourceGeneratorContext context, AdditionalText resourceFile)
+        {
+            Microsoft.CodeAnalysis.Diagnostics.AnalyzerConfigOptions options = context.AnalyzerConfigOptions.GetOptions(resourceFile);
+
+            if (options.TryGetValue("build_metadata.AdditionalFiles.GenerateSource", out string generateSourceText)
+                && !string.Equals(bool.TrueString, generateSourceText, StringComparison.OrdinalIgnoreCase))
+            {
+                // Source generation is disabled for this resource file
+                return;
+            }
+
+            if (!context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.RootNamespace", out string rootNamespace))
+            {
+                rootNamespace = context.Compilation.AssemblyName;
+            }
+
+            string resourceName = Path.GetFileNameWithoutExtension(resourceFile.Path);
+            if (options.TryGetValue("build_metadata.AdditionalFiles.RelativeDir", out string relativeDir))
```

Comment:
@chsienki This value is appearing in the generated global **.editorconfig** file, but it's not appearing in `options`

Reply:
This was fixed in https://github.com/dotnet/roslyn/pull/45357 but hasn't flowed through to any published packages yet.

Reply:
➡️ Fixed by #559